src/test/java/net/snowflake/client/jdbc/ProxyLatestIT.java

```diff
@@ -0,0 +1,346 @@
+package net.snowflake.client.jdbc;
+
+import static junit.framework.TestCase.assertEquals;
+import static net.snowflake.client.AbstractDriverIT.getConnectionParameters;
+import static net.snowflake.client.jdbc.SnowflakeUtil.systemGetProperty;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeFalse;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.ServerSocket;
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import net.snowflake.client.RunningNotOnJava8Java21;
+import net.snowflake.client.category.TestCategoryOthers;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(TestCategoryOthers.class)
+public class ProxyLatestIT {
+
+  private static final String WIREMOCK_HOST = "localhost";
+  private static final String TRUST_STORE_PROPERTY = "javax.net.ssl.trustStore";
+  private static int httpProxyPort;
+  private static int httpsProxyPort;
+  private static String originalTrustStorePath;
+  private static Process wiremockStandalone;
+
+  @BeforeClass
+  public static void setUpClass() {
+    assumeFalse(RunningNotOnJava8Java21.isRunningOnJava8Java21());
+    originalTrustStorePath = systemGetProperty(TRUST_STORE_PROPERTY);
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    System.setProperty(
+        TRUST_STORE_PROPERTY, getResourceURL("wiremock" + File.separator + "ca-cert.jks"));
+    startWiremockStandAlone();
```

Comment:
Maybe we can start wiremock only once, not per each test?

Reply:
Tests results were affected when I tried to start it only once for all tests. Maybe there is some way to clear state between tests, but operation of starting wiremock is fast, so IMHO it's low cost for having isolation between tests